src/DIRAC/WorkloadManagementSystem/Service/VirtualMachineManagerHandler.py

```diff
@@ -0,0 +1,479 @@
+    return self.virtualMachineDB.setInstanceUniqueID(instanceID, uniqueID)
+
+  types_declareInstanceSubmitted = [six.string_types]
+
+  def export_declareInstanceSubmitted(self, uniqueID):
+    """
+    After submission of the instance the Director should declare the new Status
+    """
+    return self.virtualMachineDB.declareInstanceSubmitted(uniqueID)
+
+  types_declareInstanceRunning = [six.string_types, six.string_types]
+
+  def export_declareInstanceRunning(self, uniqueID, privateIP):
+    """
+    Declares an instance Running and sets its associated info (uniqueID, publicIP, privateIP)
+    Returns S_ERROR if:
+    - instanceName does not have a "Submitted" entry
+    - uniqueID is not unique
+    """
+    gLogger.info('Declare instance Running uniqueID: %s' % (uniqueID))
+    publicIP = self.getRemoteAddress()[0]
+    gLogger.info('Declare instance Running publicIP: %s' % (publicIP))
+
+    return self.virtualMachineDB.declareInstanceRunning(uniqueID, publicIP, privateIP)
+
+  types_instanceIDHeartBeat = [six.string_types, float, six.integer_types,
+                               six.integer_types, six.integer_types]
+
+  def export_instanceIDHeartBeat(self, uniqueID, load, jobs,
+                                 transferredFiles, transferredBytes, uptime=0):
+    """
+    Insert the heart beat info from a running instance
+    It checks the status of the instance and the corresponding image
+    Declares "Running" the instance and the image
+    It returns S_ERROR if the status is not OK
+    """
+    try:
+      uptime = int(uptime)
+    except ValueError:
+      uptime = 0
+
+    res = self.virtualMachineDB.instanceIDHeartBeat(uniqueID, load, jobs,
+                                                    transferredFiles, transferredBytes, uptime)
+    return res
+
+  types_declareInstancesStopping = [list]
+
+  def export_declareInstancesStopping(self, instanceIdList):
+    """
+    Declares "Stopping" the instance because the Delete button of Browse Instances
+    The instanceID is the VMDIRAC VM id
+    When next instanceID heat beat with stopping status on the DB the VM will stop the job agent and terminates properly
+    It returns S_ERROR if the status is not OK
+    """
+    for instanceID in instanceIdList:
+      gLogger.info('Stopping DIRAC instanceID: %s' % (instanceID))
+      result = self.virtualMachineDB.getInstanceStatus(instanceID)
+      if not result['OK']:
+        return result
+      state = result['Value']
+      gLogger.info('Stopping DIRAC instanceID: %s, current state %s' % (instanceID, state))
+
+      if state == 'Stalled':
+        result = self.virtualMachineDB.getUniqueID(instanceID)
+        if not result['OK']:
+          return result
+        uniqueID = result['Value']
+        result = self.virtualMachineDB.getEndpointFromInstance(uniqueID)
+        if not result['OK']:
+          return result
+        endpoint = result['Value']
+        result = self.export_declareInstanceHalting(uniqueID, 0)
```

Comment:
Maybe you meant
```suggestion
        return self.export_declareInstanceHalting(uniqueID, 0)
```
?

Otherwise in this case we end returning `None`

Reply:
Done.